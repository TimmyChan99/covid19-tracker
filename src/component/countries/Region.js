import React from 'react';
import PropTypes from 'prop-types';
import './regions.css';
import { BsArrowRightCircle } from 'react-icons/bs';

const Region = ({ countryName, confimedCases }) => (
  <div className="the_country">
    <h3>{countryName}</h3>
    <div className="more_details">
      <span>
        {confimedCases}
        {' '}
&nbsp; new cases
      </span>

      <button
        type="button"
        value={countryName}
      >
        <BsArrowRightCircle />
      </button>

    </div>
  </div>
);
Region.propTypes = {
  countryName: PropTypes.string.isRequired,
  confimedCases: PropTypes.number.isRequired,
};
export default Region;
