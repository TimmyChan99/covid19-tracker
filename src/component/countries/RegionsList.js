import React from 'react';
import { useSelector } from 'react-redux';
import Region from './Region';

const RegionsList = () => {
  const stats = useSelector((state) => state.statsReducer);
  //console.log(stats);
  const selectedCountryName = stats[0].name;
  const selectedCountry = stats[1].filter((item) => item.countryName === selectedCountryName);
  //const { regions } = selectedCountry[0];
  const country = selectedCountry[0];
 
 console.log(country.regions);
 if (country.regions !== "no region found") {
   country.regions.map((region) => {
     
   })
 }
    
  return (

    <ul className="countries_list">
    {
      (country.regions !== "no region found") ?
      country.regions.map((region) => ( 
       <li key={region.regionName}>
        <Region 
        countryName={region.regionName}
        confimedCases={region.regionNewCases}
        />
       </li>
      )) : (<h2>no region found</h2>)
    }   
    </ul>

  );
};

export default RegionsList;
