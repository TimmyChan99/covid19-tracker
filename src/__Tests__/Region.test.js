import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from './storeMock';
import Region from '../component/countries/Region';
import RegionsList from '../component/countries/RegionsList';

it('Test Region component', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Region
        countryName="Region"
        confimedCases={152}
      />
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});

it('Test Region component', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <RegionsList />
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});
