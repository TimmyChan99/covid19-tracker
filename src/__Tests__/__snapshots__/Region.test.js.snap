// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Region component 1`] = `
<div
  className="the_country"
>
  <h3>
    Region
  </h3>
  <div
    className="more_details"
  >
    <span>
      152
       
      Â  new cases
    </span>
    <button
      type="button"
      value="Region"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
          fillRule="evenodd"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`Test Region component 2`] = `
<div
  className="message_err"
>
  <h3>
    No Statistics Found
  </h3>
  <span>
    Try again By changing the date
  </span>
</div>
`;
